persistence/src/commonMain/kotlin/com/wire/kalium/persistence/server_config/ServerConfigStorage.kt

```diff
@@ -0,0 +1,81 @@
+package com.wire.kalium.persistence.server_config
+
+import com.wire.kalium.persistence.kmm_settings.KaliumPreferences
+import com.wire.kalium.persistence.model.ServerConfigEntity
+import kotlinx.serialization.Serializable
+import kotlin.jvm.JvmInline
+
+interface ServerConfigStorage {
+    fun storeConfig(serverConfig: ServerConfigEntity): ServerConfigEntity
+    fun deleteConfig(title: String)
+    fun deleteConfig(serverConfig: ServerConfigEntity)
+    fun storedConfig(): List<ServerConfigEntity>?
+    fun serverConfigByTitle(title: String): ServerConfigEntity?
+
+    fun currentAuthServer(): ServerConfigEntity
+    fun updateCurrentAuthServer(title: String)
+}
+
+
+class ServerConfigStorageImpl(private val kaliumPreferences: KaliumPreferences) : ServerConfigStorage {
+
+    override fun storeConfig(serverConfig: ServerConfigEntity): ServerConfigEntity =
+        kaliumPreferences.getSerializable(SERVER_CONFIG_KEY, ServerConfigMap.serializer())?.data?.let { serverConfigMap ->
+            serverConfigMap[serverConfig.title] ?: run {
+                val temp = serverConfigMap.toMutableMap()
+                temp[serverConfig.title] = serverConfig
+                storeConfigMap(temp)
+                serverConfig
+            }
+        } ?: run {
+            storeConfigMap(mapOf(serverConfig.title to serverConfig))
+            serverConfig
+        }
+
+    override fun deleteConfig(title: String) {
```

Comment:
I'm kinda afraid of that considering the title of a custom backend as the key! don't know but it seems should be the base URL maybe, or the pair of title and the API base URL!